book/listings/scalable_lists/1/integer_object/mod.rs

```diff
@@ -0,0 +1,29 @@
+mod imp;
+
+use glib::Object;
+use gtk::glib;
+use gtk::prelude::*;
+
+// ANCHOR: integer_object
+glib::wrapper! {
+    pub struct IntegerObject(ObjectSubclass<imp::IntegerObject>);
+}
+
+impl IntegerObject {
+    pub fn new() -> Self {
+        Object::new(&[]).expect("Failed to create IntegerObject")
+    }
+
+    pub fn from_integer(number: i32) -> Self {
+        let integer_object = Self::new();
+        integer_object.set_property("number", number).unwrap();
+        integer_object
```

Comment:
```suggestion
        Object::new(&[("number", &number)]).unwrap();
```

And do you actually need the `pub fn new()` constructor?

Reply:
Huh, didn't know that this was possible :)
And no, in that case I don't need `new`.